src/map/script.c

```diff
@@ -21540,6 +21572,8 @@ BUILDIN(bindatcmd)
 		atcommand->binding[i]->group_lv = group_lv;
 		atcommand->binding[i]->group_lv_char = group_lv_char;
 		atcommand->binding[i]->log = log;
+		CREATE(atcommand->binding[i]->at_groups, char, db_size(pcg->db));
```

Comment:
Hmm that assertion doesn't ensure that there are no holes.

What we have right now is:

- Group ID 0: Player
- Group ID 1: Super Player
- Group ID 2: Support
- Group ID 3: Script Manager
- Group ID 4: Event Manager
- Group ID 10: Law Enforcement
- Group ID 99: Admin

There are 7 groups, but their IDs don't go from 0 to 6. And of course you can't access foo[99] in an array of [7].

There's already a function that converts a group ID to a group index though. This can make use of that. Look into `pcg->get_idx()` (I believe you only need to update `BUILDIN(add_group_command)` to make use of that, everything else is already using it. Once that's done, this line can stay as it is)